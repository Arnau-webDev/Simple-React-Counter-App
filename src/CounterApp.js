import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value }) => {

    const [ counter, setCounter ] = useState( value );

    // Handle Click
    const handleAddOnClick = (e) => { setCounter(counter + 1) };

    const handleSubstractOnClick = (e) => { setCounter(counter - 1) };

    const handleResetOnClick = (e) => { setCounter(value) };

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>

            <button onClick={ handleAddOnClick }> Add 1 to number </button>
            <button onClick={ handleResetOnClick }> Reset </button>
            <button onClick={ handleSubstractOnClick }> Minus 1 to number </button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}

CounterApp.defaultProps = {
    value: 0,
}

export default  CounterApp;
