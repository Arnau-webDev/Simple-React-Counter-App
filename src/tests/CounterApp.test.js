import React from 'react';
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

import CounterApp from "../CounterApp";

describe('Pruebas componente <CounterApp />', () => {

    let wrapper = shallow( <CounterApp />);
    let value;

    beforeEach( () => {
        value = 15;
        wrapper = shallow( <CounterApp value= { value }/>);
    })

    test("Debe renderizar el componente correctamente", () => {

        expect( wrapper ).toMatchSnapshot();
    })

    test("Debe mostrar el valor que se pasa por props correctamente", () => {

        const textToFind = wrapper.find("h2").text();

        expect( parseInt(textToFind) ).toBe( value );
    })

    test("Debe de incrementar con el boton +1", () => {

        wrapper.find("button").at(0).simulate("click");
        const textToFind = wrapper.find("h2").text();

        expect( parseInt(textToFind) ).toBe(value + 1);
    })

    test("Debe de restar con el boton -1", () => {

        wrapper.find("button").at(2).simulate("click");
        const textToFind = wrapper.find("h2").text();

        expect( parseInt(textToFind) ).toBe(value - 1);
    })

    test("Debe de resetear con el boton reset", () => {

        wrapper.find("button").at(0).simulate("click");
        wrapper.find("button").at(1).simulate("click");
        const textToFind = wrapper.find("h2").text();

        expect( parseInt(textToFind) ).toBe( value );
    })

})
