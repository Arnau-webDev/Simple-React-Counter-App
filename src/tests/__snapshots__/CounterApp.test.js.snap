// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componente <CounterApp /> Debe renderizar el componente correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    15
  </h2>
  <button
    onClick={[Function]}
  >
     Add 1 to number 
  </button>
  <button
    onClick={[Function]}
  >
     Reset 
  </button>
  <button
    onClick={[Function]}
  >
     Minus 1 to number 
  </button>
</Fragment>
`;
